### PR TITLE
Adding Selenium Code Base

### PR DESCRIPTION
**Pull request by anagarwa**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/main/java/com/adobe/cq/testing/selenium/pageobject/granite/BasePage.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+import static com.adobe.cq.testing.selenium.pagewidgets.Helpers.waitNetworkIdled;
+import static com.adobe.cq.testing.selenium.pagewidgets.Helpers.waitDocumentLoadCompleted;
+import static com.adobe.cq.testing.selenium.pagewidgets.coral.CoralReady.waitCoralReady;
+import static com.codeborne.selenide.Selenide.$;
+
+/**
+ * Base page class for inheritance.
+ */
+public class BasePage {
```
Comment (vladbailescu) [inline src/main/java/com/adobe/cq/testing/selenium/pageobject/granite/BasePage.java:40]: Shouldn't this be `abstract` if it's meant just for inheritance?

File: src/main/java/com/adobe/cq/testing/selenium/pagewidgets/common/BaseComponent.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+import static com.adobe.cq.testing.selenium.pagewidgets.Helpers.waitForElementAnimationFinished;
+import static com.adobe.cq.testing.selenium.pagewidgets.coral.CoralReady.waitCoralReady;
+import static com.adobe.cq.testing.selenium.utils.ElementUtils.clickableClick;
+import static com.codeborne.selenide.Selenide.$;
+
+/**
+ * Base component class for inheritance.
+ */
+public class BaseComponent {
```
Comment (vladbailescu) [inline src/main/java/com/adobe/cq/testing/selenium/pagewidgets/common/BaseComponent.java:37]: Shouldn't this be `abstract`?

**Comment by anagarwa** [comment]
@vladbailescu @tyge68 @absarasw can you please help review this pr. 

thanks,

File: src/main/java/com/adobe/cq/testing/selenium/pageobject/granite/BasePage.java (hunk 1 above)
Comment (anagarwa) [reply src/main/java/com/adobe/cq/testing/selenium/pageobject/granite/BasePage.java]: updated

File: src/main/java/com/adobe/cq/testing/selenium/pagewidgets/common/BaseComponent.java (hunk 2 above)
Comment (anagarwa) [reply src/main/java/com/adobe/cq/testing/selenium/pagewidgets/common/BaseComponent.java]: updated

**Review by vladbailescu — APPROVED** [review]

**Pull request merged**